dbfit-java/core/src/main/java/dbfit/api/DbStoredProcedureCall.java

```diff
@@ -51,7 +51,12 @@ void bindParametersTo(StatementExecution cs) throws SQLException {
     }
 
     public StatementExecution toStatementExecution() throws SQLException {
-        StatementExecution cs = new StatementExecution(this.environment.getConnection().prepareCall(toSqlString()));
+        StatementExecution cs;
+        if (isFunction()) {
+            cs = environment.createFunctionStatementExecution(this.environment.getConnection().prepareCall(toSqlString()), true);
```

Comment:
In general I'd vote for the sub-classing option for now. I just spotted some specifics - I'll comment inline.